### DISCUSSION TITLE
How to remove the current page from a list of pages?

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#795] [Ask a question] [answered]
Dears,

this is related to discussion datenstrom/community#455 from 2019, but I need help on implementing the proposed solution (if it is still valid).

Just like datenstrom/community#455, I have a blog page and I want to include a section linking to related blogposts. 
The blog pages are grouped by a common tag (e.g. tag:coffee). 
As shortcode, I either use [blogpages /blog/ coffee] or [blogchanges /blog/ coffee].

My goal is to "filter out" and hide the current page from this link list.

I have seen the solution using (diff), but my PHP is not sufficient enough: I am not sure where and how to exactly implement this (in blog.php? in core.php? elsewhere?)

Any help appreciated, as always!

Cheers,

Carsten

**Comment by JetGum** [comment]
Dears,

Unfortunately, I can't program this myself, as my PHP knowledge is not sufficient ...

Maybe someone has had a similar problem and solved it before? I'm sure I'm not the only one who runs a blog and assigns tags to blog posts, using the blogpages/blogchanges syntax.

Maybe there is a creative approach to using the tags without changing the programming?

Cheers, Carsten

**Comment by schulle4u** [comment, marked as answer]
This is now possible. Here is an example snippet to remove the current page from a page collection: 

```
$pages = $this->yellow->content->index()->filter("layout", "blog");
$pages->remove($this->yellow->page);
```

Does this solve your problem?

**Comment by JetGum** [comment]
Apologies, but could you please be specific where to put the code in blog.php and/or what lines of code to replace? Then I can test.
I am using the blogpages as well as the blogchanges shortcut.

Cheers, Carsten

**Reply by JetGum** [reply]
Ah, I think I got it. 

I updated core and blog via php yellow.php update all, then added the 2 lines in blog.php underneath // Retrun blogpages shortcust, right after '$pages = $this->getBlogPages($blogStart);'

Works like a charm.

Question: Will these 2 lines overwritten when I update blog next time?

Thanks a million!

**Reply by schulle4u** [reply]
Yes, as this is the main extension file it will be owerwritten upon next update. Alternatively you could create your own blog shortcut extension and use it in your layout file, or simply recreate the code in your layout. Have a look at datenstrom/yellow#842 and referenced discussions for code examples, I'm pretty sure you'll find something useful for your needs. 🙂

**Reply by markseuffert** [reply]
Now the `[blogpages]` shortcut doesn't show the current page anymore.

In other places it's like this for a long time. For example I use the [wiki extension](https://github.com/annaesvensson/yellow-wiki) and the `[wikipages]` shortcut to show a list of all articles on the main page. [Screenshot for the curious](https://user-images.githubusercontent.com/2429165/231087177-42b6c1c5-1507-4c40-8a9d-0fc20fcab994.png). It wouldn't make sense to show the main page here again. Another example is `pages->similar()` in the API, which gives you a page collection with pages similar to the current page. The page collection doesn't include the current page. It think it's good to have the same behaviour in the blog extension.
